src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -485,22 +485,22 @@ public IOpenApiReferenceable ResolveReference(OpenApiReference reference)
         /// <param name="referenceUri"></param>
         /// <returns>A JsonSchema ref.</returns>
         public JsonSchema ResolveJsonSchemaReference(Uri referenceUri)
-        {            
+        {
+            const char pound = '#';
             string uriLocation;
-            string id = referenceUri.OriginalString.Split('/')?.Last();
-            string relativePath = "/components/" + ReferenceType.Schema.GetDisplayName() + "/" + id;
-            
-            if (referenceUri.OriginalString.StartsWith("#"))
+            int poundIndex = referenceUri.OriginalString.IndexOf(pound);
+
+            if (poundIndex > 0)
```

Comment:
Just confirming, this looks to be functially equivalent to the previous state of  `if (referenceUri.OriginalString.StartsWith("#"))`.
Any reason for the change as `poundIndex` isn't used elsewhere?

Reply:
In the current implementation, I'm specifically checking that:
1. The `referenceUri` string value has a # present, and
2. The # is not the 1st character in the string value

In the previous example we were checking whether the # character is the 1st character in the  `referenceUri` string value.

Reply:
Thanks for clarifying this!